File: app/src/main/java/com/example/kirchhoff/example/ui/animation/CustomAdapter.java
package com.example.kirchhoff.example.ui.animation;

import android.content.Context;
import android.graphics.Color;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.LinearLayout;
import android.widget.TextView;

import com.example.kirchhoff.example.R;

import java.util.ArrayList;
import java.util.List;

/**
 * @author Kirchhoff-
 */

public class CustomAdapter extends BaseAdapter {

    private ArrayList<String[]> arrayList;
    private Context context;

    public CustomAdapter(Context context, List<String[]> list) {
        this.context = context;
        arrayList = new ArrayList<>(list);
    }

    @Override
    public int getCount() {
        return arrayList.size();
    }

    @Override
    public Object getItem(int position) {
        return arrayList.get(position);
    }

    @Override
    public long getItemId(int position) {
        return position;
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        View row;
        LayoutInflater inflater = (LayoutInflater) context
                .getSystemService(Context.LAYOUT_INFLATER_SERVICE);
        if (convertView == null) {
            row = inflater.inflate(R.layout.r_simple_transition, parent,
                    false);
        } else {
            row = convertView;
        }
        String[] detail = arrayList.get(position);


        LinearLayout layout = (LinearLayout) row.findViewById(R.id.layout);
        layout.setBackgroundColor(Color.parseColor(detail[3]));
        TextView tvName = (TextView) row.findViewById(R.id.tvName);
        tvName.setText(detail[0]);
        TextView tvEmail = (TextView) row.findViewById(R.id.tvEmail);
        tvEmail.setText(detail[1]);

        return row;
    }

}

File: simpletransition/src/main/java/com/example/simpletransition/CustomAdapter.kt
package com.example.simpletransition

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.LinearLayout
import android.widget.TextView
import androidx.core.content.ContextCompat

internal class CustomAdapter(private val context: Context, list: List<TransitionItem>) : BaseAdapter() {

    private val arrayList = ArrayList(list)

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        val row: View
        val inflater = context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
        row = convertView ?: inflater.inflate(R.layout.r_simple_transition, parent, false)
        return createRow(row, getItem(position))
    }

    override fun getCount() = arrayList.size

    override fun getItem(position: Int): TransitionItem = arrayList[position]

    override fun getItemId(position: Int) = position.toLong()

    private fun createRow(row: View, item: TransitionItem): View {
        val layout = row.findViewById<LinearLayout>(R.id.layout)
        layout.setBackgroundColor(ContextCompat.getColor(context, item.color))
        val tvName = row.findViewById<TextView>(R.id.tvName)
        tvName.text = item.title
        val tvLanguage = row.findViewById<TextView>(R.id.tvLanguage)
        tvLanguage.text = item.language
        return row
    }
}
